Implementation for new SMT ExtractTopicFromValueSchema and tests (#93)
aiven/transforms-for-apache-kafka-connect: README.md

Only, `string`, `numeric` and `boolean` types are considered for matching purposes, other types are ignored.

### `ExtractTopicFromValueSchema`

This transformation checks the record value schema name and if it exists uses it as the topic name.

- `io.aiven.kafka.connect.transforms.ExtractTopicFromValueSchema$Name` - works on value schema name.

> By default the transformation uses the content of the schema.name field of the record value.

Reviewer: ```suggestion
By default the transformation uses the content of the schema.name field of the record value and uses the first matched group.
```

This would imply changing the example regex to give single matched group. The current regex gives to in Java.

The transformation defines the following configurations which can be used to tamper the schema.name:

- `SchemaNameToTopicMap` - Map that contains the schema.name value and corresponding new topic name value that should be used instead. Format is "SchemaValue1:NewValue1,SchemaValue2:NewValue2" so key:value pairs as comma separated list.

> - `RegExSchemaNameToTopicMap` - RegEx that should be used to parse the schema.name to desired value. For example for example (?:[.\t]|^)([^.\t]*)$ which parses the name after last .

Reviewer: ```suggestion
- `RegExSchemaNameToTopicMap` - RegEx that should be used to parse the schema.name to desired value. For example for example (?:[.\t]|^)([^.\t]*)$ which parses the name after last dot
```

And would the regex `(?:[.]|^)([^.]*)$` be enough?

> - `schema.name.regex` - RegEx that should be used to parse the schema.name to desired value. For example for example (?:[.]|^)([^.]*)$ which parses the name after last dot.

Reviewer: ```suggestion
- `schema.name.regex` - RegEx that should be used to parse the schema.name to desired value. For example for example `(?:[.]|^)([^.]*)$` which parses the name after last dot.
```

> Here is an example of this transformation configuration (using :SchemaNameToTopicMap)

Reviewer: ```suggestion
Here is an example of this transformation configuration (using :schema.name.topic-map)
```

```properties
transforms=ExtractTopicFromValueSchema
transforms.ExtractTopicFromValueSchema.type=io.aiven.kafka.connect.transforms.ExtractTopicFromValueSchema$Name
transforms.ExtractTopicFromValueSchema.schema.name.topic-map=com.acme.schema.SchemaNameToTopic1:TheNameToReplace1,com.acme.schema.SchemaNameToTopic2:TheNameToReplace2

```

> 

Reviewer: ```suggestion

And here is an example of this transformation configuration (using :schema.name.regex)

```properties
transforms=ExtractTopicFromValueSchema
transforms.ExtractTopicFromValueSchema.type=io.aiven.kafka.connect.transforms.ExtractTopicFromValueSchema$Name
transforms.ExtractTopicFromValueSchema.schema.name.regex=(?:[.]|^)([^.]*)$
```
```
